### PR TITLE
Lint fixes

### PR DESCRIPTION
**Pull request by trusktr**
This applies lint fixes after config updates in https://github.com/nasa/openmct/pull/4554. #4554 is the PR that matters. When we're ready to merge, we need to re-do this PR as it will most likely have conflicts after anything is merged to master.

So when we're ready to merge #4554, then we should re-make this and merge this at the same time.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [ ] Command line build passes?
* [ ] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

Testing: The `npm run lint` command should pass

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by trusktr** [comment]
@unlikelyzero looks like node10-lint failed. It is working fine for me locally on Node 17.

**Comment by trusktr** [comment]
Any idea why the other tests failed?

**Comment by unlikelyzero** [comment]
@trusktr I don't see anything obvious and can't think of why these tests could be impacted by the linting changes

**Comment by trusktr** [comment]
@unlikelyzero When I run local build, I see `inMemorySearchWorker.js` in the `dist/` folder.

**Comment by unlikelyzero** [comment]
@trusktr it's possible this override is affecting it https://github.com/nasa/openmct/blob/92bb687e1bbf016131de3c77540ef84d922e61cc/.circleci/config.yml#L6-L7

**Comment by trusktr** [comment]
@unlikelyzero Oops, my mistake, I change `prepare` to `xprepare` to avoid having to build locally whenever `npm install` runs, for speed.

**Comment by trusktr** [comment]
Do we have to keep node10 support?

**Comment by unlikelyzero** [comment]
For now, yea. Might need to temp disable the ployfill rule and then make a comment for removing that rule when we formally remove support for 10.

**Comment by trusktr** [comment]
Oh ok. It occurred to me, do we need to support node 10 for linting? Maybe we can lint any way we want (including with new Node), as it isn't really a thing for consumers.

**Comment by trusktr** [comment]
Looks like node10-lint (which is currently using Node 17) fails because it hangs on the webpack command although webpack is finished building.

**Comment by unlikelyzero** [comment]
@trusktr i think we need to support node10 without purposefully breaking it until we can confirm that VISTA will build on node12. VIPER will build on node18. I don't believe that we need to _test_ it. We're currently only running linting on node10 because it was part of our original support matrix and we need to run linting on each PR.

**Review by unlikelyzero — APPROVED** [review]
This is good to go now that we've moved off of node10

**Comment by akhenry** [comment]
This requires a corresponding issue. Can someone give some context about what this PR is fixing?

**Comment by trusktr** [comment]
@akhenry issue: https://github.com/nasa/openmct/issues/4659

The PR updated eslint dependencies, which required some lint fixes across the code base.

**Comment by trusktr** [comment]
I need to make sure this will pass using a newer version of Node before we can merge.

**Comment by unlikelyzero** [comment]
@trusktr you should pull in the changes to circle.yml

**Comment by shefalijoshi** [comment]
@davetsay Please take a look at this PR. Thanks!

**Review by davetsay — CHANGES_REQUESTED** [review]
Looks good. My one change would be to remove the `eslint-disable-next-line` comment and fix the test instead, unless what i've suggested breaks the intent of the test.

File: platform/telemetry/test/TelemetryFormatterSpec.js
Comment (jvigliotta) [reply platform/telemetry/test/TelemetryFormatterSpec.js:60]: @trusktr Can you take a look at look at what Dave is asking here?

File: .eslintrc.js
Comment (trusktr) [reply .eslintrc.js]: Yes, this causes errors now after the update. Official recommendation is to follow Vue conventions. This is thread is now stale, I have updated it to `off` instead (in the update-eslint PR), and now these are skipped, and I added a TODO to do them later. This makes things work like they did before, but we're not following conventions.

File: platform/telemetry/test/TelemetryFormatterSpec.js
Comment (trusktr) [reply platform/telemetry/test/TelemetryFormatterSpec.js]: My intent was not to change any runtime behavior.

However, I've since updated the base update-eslint PR, and I re-made this PR, and `lint:fix` passed without me needing to add any comment or change any code.

**Review by unlikelyzero — APPROVED** [review]
I would wait until we can get the reference PR merged and then get 2 of the reviewers on a call to immediately rebase and actively merge this PR. Simply due to the number of files changed and file conflicts

**Comment by trusktr** [comment]
Agreed. Also the main change is that the recommended Vue lint setting no longer causes the weird HTML attribute spacing, which is why the changes are most re-indentation of the HTML attributes (way better now).

But, maybe we can configure a setting to restore the old behavior. Would that be preferable?

**Comment by unlikelyzero** [comment]
@trusktr i think it would be ok to fix it once and for all. Also, how would this be impacted by any prettier enforcement? Maybe we could include the HTML Attribute spacing change at the same time that we implement prettier?

**Comment by trusktr** [comment]
@unlikelyzero  Prettier uses the same formatting as this new one in ESLint for attribute spacing. Running Prettier on the code base will introduce many more changes too.

I can make a PR to show what it looks like.

**Comment by trusktr** [comment]
I'll make the Prettier PR build off of this one, then make a second formatting PR to show what that looks like

**Review by davetsay — APPROVED** [review]
LGTM

## Reviewer Checklist
| | |
| --- | :---: |
| Changes appear to address issue? | Y |
| Appropriate unit tests included? | N/A |
| Code style and in-line documentation are appropriate? | Y |
| Commit messages meet standards? | Y |
| Has associated issue been labelled `unverified`? | N/A |

**Pull request merged**
